### PR TITLE
V2 samples Spring Security + Okta Starter

### PR DESCRIPTION
**Pull request by bdemers**
NOTE: the custom-login is broken ATM, i'll fix this, but I wanted to get it in your hands first @vijetmahabaleshwar-okta (just use the `basic` example)

### CODE REVIEW

**Comment by bdemers** [comment]
@vijetmahabaleshwar-okta take a look!

File: resource-server/src/main/java/com/okta/spring/example/ResourceServerExampleApplication.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
+package com.okta.spring.example;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.autoconfigure.security.oauth2.resource.ResourceServerProperties;
```
Comment (vijetmahabaleshwar-okta) [inline resource-server/src/main/java/com/okta/spring/example/ResourceServerExampleApplication.java]: nit: unused imports in the file - 

ResourceServerProperties
ResourceServerSecurityConfigurer
RequestMethod

File: README.md [hunk 2]
```diff
@@ -1,457 +1,76 @@
...
-Start the back-end for your sample application with `npm start` or `mvn -f lib/pom.xml spring-boot:run --quiet`. This will start the app server on [http://localhost:3000](http://localhost:3000).
+### `custom-login`
 
-By default, this application uses a mock authorization server which responds to API requests like a configured Okta org - it's useful if you haven't yet set up OpenID Connect but would still like to try this sample. 
-
-To start the mock server, run the following in a second terminal window:
 ```bash
-# Starts the mock Okta server at http://127.0.0.1:7777
-[samples-java-spring-mvc]$ npm run mock-okta
+cd basic
```
Comment (vijetmahabaleshwar-okta) [inline README.md]: change to `cd custom-login`

File: README.md [hunk 3]
```diff
@@ -1,457 +1,76 @@
...
-In this sample, we use AuthJS to create the URL and perform the redirect. An `OktaAuth` object is instantiated with the configuration in `.samples.config.json`. When the `login()` function is called from the view, it calls the [`/authorize`](http://developer.okta.com/docs/api/resources/oauth2.html#authentication-request) endpoint to start the [Authorization Code Flow](https://tools.ietf.org/html/rfc6749#section-1.3.1).
-
-You can read more about the `OktaAuth` configuration options here: [OpenID Connect with Okta AuthJS SDK](http://developer.okta.com/code/javascript/okta_auth_sdk#social-authentication-and-openid-connect).
-
-**Important:** When the authorization code is exchanged for an `access_token` and/or `id_token`, the tokens **must** be [validated](#validation). We'll cover that in a bit.
+front-end:
 
-### Custom Login Form
-To render the [Okta Sign-In Widget](http://developer.okta.com/code/javascript/okta_sign-in_widget), include a container element on the page for the widget to attach to:
+This project is just serving a static `index.html` so you WILL need to update the configuration in that file, see line 53:
```
Comment (vijetmahabaleshwar-okta) [inline README.md]: It'd be nice to provide the exact path to index.html in front-end directory, namely `front-end/src/main/resources/static/index.html`

File: README.md [hunk 4]
```diff
@@ -1,457 +1,76 @@
...
-        responseType: 'code',
-        scopes: ['openid', 'email', 'profile'],
-      },
-    });
-    signIn.renderEl({ el: '#sign-in-container' }, () => {});
+```javascript
+const data = {
+  baseUrl: 'https://dev-123456.oktapreview.com',
+  clientId: 'YourClientId',
+  redirectUri: 'http://localhost:8080',
```
Comment (vijetmahabaleshwar-okta) [inline README.md]: redirectUri for SPA is `http://localhost:8080/implicit/callback` by default
Also, should we mention that the user needs to have an SPA created in okta to test resource server flow?

Comment (bdemers) [reply README.md]: `implicit/callback` doesn't work in this case as it is only a static file.
This front end piece will be replaced with one of the front end libs once those are in place

**Pull request merged**